src/ProjectTemplates/Web.ProjectTemplates/content/Grpc-WebApplication/Program.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Company.WebApplication1
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            CreateHostBuilder(args).Build().Run();
+        }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.ConfigureKestrel(options =>
+                    {
+                        options.Limits.MinRequestBodyDataRate = null;
+                        options.ListenLocalhost(50051, listenOptions =>
```

Comment:
Maybe something like this instead of hard coding the ports:

```C#
webBuilder.ConfigureKestrel(options =>
{
    options.ConfigureEndpointDefaults(listenOptions  =>
    {
        listenOptions.Protocols = HttpProtocols.Http2;
    });
});
```

Just default all endpoints to HTTP2. You may also want to consider hiding this in the call to AddGrpc

Reply:
I'd like to avoid hard coding a port too.

If we do go with a dynamic port, and we have a multi project template with a client and a server, how do we ensure the client calls the server on the correct port?

Reply:
I thought extension methods off ServiceCollection were just for services. Are there other libraries that make calls to configure Kestrel in the add services extension method?

Reply:
> If we do go with a dynamic port, and we have a multi project template with a client and a server, how do we ensure the client calls the server on the correct port?

Lets use configuration to set the protocol to HTTP2 then we can discuss if it's possible to do anything with the client and server. I don't know of a good way today.

> I thought extension methods off ServiceCollection were just for services. Are there other libraries that make calls to configure Kestrel in the add services extension method?

You add an `IConfigureOptions<KestrelServerOptions>`, it does mean you need a dependency on Kestrel. I don't know what we should do here but I haven't given it much thought as yet.